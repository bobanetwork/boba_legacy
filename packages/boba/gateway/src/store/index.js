/*
Copyright 2021-present Boba Network.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License. */

import { createStore , applyMiddleware } from 'redux'
import RootReducer from 'reducers'
import reduxThunk from 'redux-thunk'
import persistReducer from 'redux-persist/lib/persistReducer'
import storage from 'redux-persist/lib/storage'
import persistStore from 'redux-persist/lib/persistStore'

const initialState = {}

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['network']
}

const persistedReducer = persistReducer(persistConfig, RootReducer)

export const store = createStore(
  persistedReducer,
  initialState,
  applyMiddleware(reduxThunk)
)

export default store;

export const persistor = persistStore(store)
