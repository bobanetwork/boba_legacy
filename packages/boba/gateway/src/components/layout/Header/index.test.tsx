import { render, screen } from '@testing-library/react'
import React from 'react'
import { Header } from '.'
import CustomThemeProvider from 'themes'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import store from 'store'

const renderHeader = () => {
  return render(
    <MemoryRouter>
      <Provider store={store}>
        <CustomThemeProvider>
          <Header />
        </CustomThemeProvider>
      </Provider>
    </MemoryRouter>
  )
}

describe('Layout => Header', () => {
  test('should match the snapshot', () => {
    const { asFragment } = renderHeader()

    expect(asFragment()).toMatchSnapshot()
  })

  test('should have button with label connect wallet', () => {
    renderHeader()
    expect(screen.getByText('Connect Wallet')).toBeVisible()
  })
})
