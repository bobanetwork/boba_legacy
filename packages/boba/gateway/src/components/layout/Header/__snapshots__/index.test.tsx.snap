// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout => Header should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-fABZjn fNXwka"
  >
    <a
      class=""
      href="/bridge"
    >
      <div
        class="sc-gUMnWI dkUFbU"
      />
    </a>
    <div
      class="sc-hjLWnZ kFZkac"
    >
      <a
        activeclassname="active"
        class="sc-fWFeAW dJFBmS"
        href="/bridge"
      >
        Bridge
      </a>
      <a
        activeclassname="active"
        class="sc-fWFeAW dJFBmS"
        href="/history"
      >
        History
      </a>
      <a
        activeclassname="active"
        class="sc-fWFeAW dJFBmS"
        href="/earn"
      >
        Earn
      </a>
      <a
        activeclassname="active"
        class="sc-fWFeAW dJFBmS"
        href="/stake"
      >
        Stake
      </a>
      <a
        activeclassname="active"
        class="sc-fWFeAW dJFBmS"
        href="/DAO"
      >
        Dao
      </a>
    </div>
    <div
      class="sc-hUOJWJ kkAMsz"
    >
      <button
        class="sc-crozmw joSqYT"
        label="Connect Wallet"
        style="white-space: nowrap;"
        type="button"
      >
         Connect Wallet
      </button>
      <div
        class="sc-dPwPAC UluVn"
      >
        <div
          class="sc-ePzlA-D hiWbI"
        >
          <div>
            <div
              data-src="hamburger.svg"
              src=""
            />
          </div>
        </div>
      </div>
      <div
        class="sc-gjJAnY imSQnB"
        title="light-icon"
      >
        <div>
          <div>
            <div
              data-src="theme-sun.svg"
              src=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
